### PR TITLE
Add math to lch fields #74

### PR DESCRIPTION
**Pull request by raidenmiro**
## Description (WIP)

Example of implementation #74

## Checklist

- [ ] - Show `spinbutton` on desktop only (?)
- [ ] - Test this component on a screen reader ([w3.org](https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/))
- [ ] - Button highlighting by pressing and holding with the mouse and moving by Y
- [x] - Add `inputmode` for input
- [x] - Support keyboard `ArrowUp`, `ArrowDown`, `Home`, `End`
- [X] - Support for holding the button and `increasing` or `decreasing` the value
- [x] - Support for increasing or decreasing the value when holding down and changing the position of the cursor by Y
- [x] - Validating the value, allowing only numbers, `.` and operators `+, -, /, *` (only numbers are allowed in the `alpha` field).

I would welcome suggestions on what can be fixed or explained

### CODE REVIEW

File: lib/math.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+let LowPriority = 1
```
Comment (ai) [inline lib/math.ts:1]: TS Enum?

File: view/field/mixin.pug [hunk 2]
```diff
@@ -2,6 +2,12 @@ mixin field(label, hotkey, opts = {})
   .field( class=opts.options ? 'is-options' : '' )
     kbd.field_hotkey= hotkey
     input.field_input( aria-label=label )&attributes({ type: 'text', ...opts })
+    if (opts.role === 'spinbutton')
+      .controls
+        button.control.increase( tabindex="-1" )
```
Comment (ai) [inline view/field/mixin.pug:7]: Modifiers should be in `is_increase` and elements should have a selector like `BLOCK_ELEMENT` (`.field_control`).

File: lib/math.ts (hunk 1 above)
Comment (raidenmiro) [reply lib/math.ts:1]: Good point! I'll fix it.

File: view/field/mixin.pug (hunk 2 above)
Comment (raidenmiro) [reply view/field/mixin.pug]: Done

**Comment by ai** [comment]
Could you increase size limit in the config?

**Comment by raidenmiro** [comment]
Sorry for the delay 😅

File: .size-limit.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
       "!dist/*.webmanifest",
       "!dist/*.png"
     ],
-    "limit": "60 KB"
+    "limit": "60.57 KB"
```
Comment (ai) [inline .size-limit.json:11]:
```suggestion
    "limit": "65 KB"
```

I am using 5 KB step for limit in this project

File: lib/math.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  '-': (a: number, b: number) => b - a,
+  '*': (a: number, b: number) => a * b,
+  '/': (a: number, b: number) => b / a
+}
+
+function isContinuedDigit(ch: string): boolean {
+  return /^[0-9]*\.?[0-9]*$/.test(ch)
+}
+
+function getPriority(ch: keyof Operators): number {
```
Comment (ai) [inline lib/math.ts:26]: Why do we need it?

File: lib/math.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  for (let ch of postfixOfExpression) {
+    if (ch in operations) {
+      let callback = operations[ch as keyof Operators]
+      stack.push(callback(stack.pop()!, stack.pop()!))
+    } else {
+      stack.push(parseValue(ch))
+    }
+  }
+
+  return stack.pop() ?? DefaultValue
```
Comment (ai) [inline lib/math.ts:90]: Let’s inline this constant

File: view/field/mixin.pug [hunk 6]
```diff
@@ -2,6 +2,12 @@ mixin field(label, hotkey, opts = {})
   .field( class=opts.options ? 'is-options' : '' )
     kbd.field_hotkey= hotkey
     input.field_input( aria-label=label )&attributes({ type: 'text', ...opts })
+    if (opts.role === 'spinbutton')
+      .field_controls
+        button.field_control.is_increase( tabindex="-1" )
```
Comment (ai) [inline view/field/mixin.pug:7]:
```suggestion
        button.field_control.is-increase( tabindex="-1" )
```

File: lib/math.ts (hunk 4 above)
Comment (raidenmiro) [reply lib/math.ts:26]: :thinking: Regular expression is unnecessary, here we merely check if the current character is a continuation of a number (It is need to check if the last element is a number or a point). I can create such a check (without additional function)

```ts
let isContinuedNumber = /^[\d.]+$/.test(str.charAt(i - 1)) 
```

`getPriority` - some wrapper, for more readable then `operators[currentOpeator]`

File: lib/math.ts (hunk 5 above)
Comment (ai) [reply lib/math.ts]: @raidenmiro the last thing to fix

File: view/field/mixin.pug (hunk 6 above)
Comment (raidenmiro) [reply view/field/mixin.pug]: Oh, my bad

**Pull request merged**